Add navbar and first section to page.

poncem91 opened the pull request:
This adds the barebones of the portfolio page, including a navbar and a first section that includes a header, paragraph, and picture.

poncem91 committed:
Add navbar and first section to page.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -2,14 +2,33 @@
+          It wasn't until after I graduated from college that I discovered my passion for Computer Science and programming, 
+          so I decided to go back to school to complete a second bachelors in Computer Science. 
+          I just wrapped up my first year as a CS student and I am loving every minute of it!
+      <br><br>
```
Rather than inserting breaks, can you close the paragraph block here and start a new one for the picture of Arlo? That's more semantic, and it will give you a similar kind of separation on the page, which you can adjust with CSS if desired.

poncem91 committed:
Replace <br> tags, add more content, style further

Commit gets rid of <br> tags and uses <p> tags instad.
More site content is added in the HTML file, which has been divided into sections styled in CSS.
Small changes made to the button message in JS.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
-            <a href="#aboutme">About Me</a>
-            <a href="#link2">Link 2</a>
-            <a href="#link3">Link 3</a>
+            <a href="#void">About Me</a>
```
What's `#void`? Shouldn't these be `#aboutme` for the About Me section, `#projects` for Projects, etc.?

poncem91 replied:
I wanted to use JS to make the transitions in the links, but I haven't implemented that yet so I used #void as a dead link placeholder. In the future should I remove the link altogether or what is a good way to set placeholders for implementations that will come with the next commit?
I'll make sure to have this fixed with the next commit though!

peap replied:
A plain `#` sign will do as a placeholder, though I would try to avoid adding placeholders, preferring to add functionality only when you need it. For example with these links, it would be better to keep them functioning by referencing the actual headers they refer to until the time that you get the smooth scrolling feature added.

poncem91 replied:
This makes perfect sense, thanks!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
+              <li><b>Finance:</b> A mock stock-trading website implemented using Python, SQL, and Flask, where a user can register, login, change passwords, buy and sell stocks, and see their transaction history.</li>
+              <li><b>Liberty Slot Machine:</b> A three-part program implemented in Python to simulate a game of the Liberty Bell Slot Machine.
+              <ol>
+                  <li>Tests a serious of known spins and outputs expected payout.</li>
```
> serious

series?

poncem91 replied:
Typo, will fix, thanks for catching that!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
-      <p>Hello! My name is Mafe and I am currently a STEP intern at Google. 
+
+      <!-- About Me Section -->
+      <div class="section" id="aboutme">
```
HTML has a dedicated `section` tag that would be good to use for your sections.

poncem91 replied:
Thanks! I didn't know that there was, that'll work much better.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
   </head>
   <body>
+
+    <!-- Sticky Navbar -->
```
Happy to see these comments here! It definitely helps you keep the context in mind when scrolling through files written in relatively verbose languages like HTML.

poncem91 replied:
Thanks!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
+          </p>
+          
+          <!-- Arlo Sub-Section -->
+          <center>
```
Probably better to put this in a `div` with an ID that you can reference to center it in CSS.

poncem91 replied:
Sounds good! I'll make sure to move all stylizing (bold, center, italics, etc.) into CSS.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
+          <!-- Arlo Sub-Section -->
+          <center>
+              <img src="/images/arlo.jpg" alt="My dog Arlo">
+              <p><i>Click here to make Arlo talk:</i></p>
```
Same here: probably better to italicize this paragraph in CSS.

But it's good to see Arlo talking here!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
+      <div class="section" id="projects">
+          <h1>Projects</h1>
+          <ul>
+              <li><b>Finance:</b> A mock stock-trading website implemented using Python, SQL, and Flask, where a user can register, login, change passwords, buy and sell stocks, and see their transaction history.</li>
```
Prefer to use CSS over styling tags. See the `em` and `strong` tags, which impart semantic meaning to their contents. Alternatively, we have the description list tag `dl`, which seems suited to this kind of listing: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl.

poncem91 replied:
Thanks this is super helpful!

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -7,31 +7,71 @@
+          <ul>
+              <li><b>Finance:</b> A mock stock-trading website implemented using Python, SQL, and Flask, where a user can register, login, change passwords, buy and sell stocks, and see their transaction history.</li>
+              <li><b>Liberty Slot Machine:</b> A three-part program implemented in Python to simulate a game of the Liberty Bell Slot Machine.
+              <ol>
```
It's up to you, but thought I'd check: did you intend to put an ordered list in the middle of an unordered list?

poncem91 replied:
I did, but I think I'll change it to unordered since it does look a bit odd.

peap commented on portfolio/src/main/webapp/style.css:
```diff
@@ -1,30 +1,46 @@
+    margin-bottom: 5px;
 }
 
+/* Sticky Navbar */
```
I like how this navbar looks on the page - nice!

poncem91 replied:
Thanks!!

peap commented on portfolio/src/main/webapp/style.css:
```diff
@@ -33,29 +49,34 @@ p {
     width: 100%;
     z-index: 100;
-    line-height: 80px;
+    transition: 0.5s;
```
What transition is this? Something to do with clicking the links?

poncem91 replied:
My bad, this shouldn't be there. This is a transition for my shrink nav bar JS function, but I wanted to wait until these current changes had been reviewed first before including the function in my next commit.

poncem91 committed:
Fix requested changes, add shrink navbar functionality, add smooth scrolling to anchors

Commit makes use of <section> tags to divide site sections and moves all stylizing to CSS.
Using JS, shrink navbar functionality makes the navbar shrink when scrolling down and go back to its normal size when scrolling back to top of page.
Commit also implements the navbar links and uses smooth scrolling to scroll to the reach each section of the page.

peap commented on portfolio/src/main/webapp/index.html:
```diff
@@ -14,18 +14,18 @@
-            <a href="#void">Projects</a>
-            <a href="#void">Work Experience</a>
-            <a href="#void">Contact Me</a>
+            <a href="#" onclick="scrollToAboutMe();return false">About Me</a>
```
Can you make these click functions return false, instead of adding that as a second statement here?

poncem91 replied:
Yes, this is a good idea, thanks!

peap commented on portfolio/src/main/webapp/script.js:
```diff
@@ -26,3 +26,71 @@ function addRandomArloMessage() {
+/**
+ * Smooths scrolls to About Me Section / top of document
+ */
+function scrollToAboutMe() {
```
Can you combine these four functions into a single one that takes the element to scroll to, and perhaps an additional offset, as arguments, then pass those arguments in the HTML `onclick` attributes? That would remove the repetition in this file and make it easier to maintain. I see that `scrollToAboutMe` uses `window.pageYOffset`, but I suspect it's more-or-less a constant offset away from `document.getElementById("aboutme").getBoundingClientRect().top`, which makes it look like the others.

poncem91 replied:
I really appreciate all these suggestions, this does makes things a lot cleaner.

poncem91 committed:
combine four scroll functions and remove second statement on onclick property

Commit combines four scroll functions into one by passing the section id as a function argument. The single function now also returns false to avoid a second statement inside the onclick HTML event. Additionally I return the scrollToSection function in the onclick event, since not returning it causes the scroll functionality to toggle up and down when the navbar links are clicked more than once.
